src/uri.rs

```diff
@@ -0,0 +1,261 @@
+    items(ItemId),
+    item_stats(Option<ItemId>),
+    // this is nothing to do with a specific item.
+    // item stats supporst 3 end points/
+    // base /itemstats
+    // id   /itemstats/id
+    // multiple(?) /itemstats?ids=23
+    item_stats_all(Option<ItemStatId>),
+    recipes(Option<RecipeId>),
+    build,
+}
+
+impl EndPoint {
+    pub fn requires_auth(self) -> bool {
+        match self {
+            // do require auth.
+            EndPoint::account => true,
+            EndPoint::account_achievements => true,
+            EndPoint::account_bank => true,
+            EndPoint::account_materials => true,
+            EndPoint::account_dailycrafting => true,
+            EndPoint::account_dungeons => true,
+            EndPoint::account_dyes => true,
+
+            // don't require auth
+            EndPoint::achievements(_) => false,
+            EndPoint::achievements_daily => false,
+            EndPoint::achievements_daily_tomorrow => false,
+            EndPoint::items(_) => false,
+            EndPoint::item_stats(_) => false,
+            EndPoint::item_stats_all(_) => false,
+            EndPoint::recipes(_) => false,
+            EndPoint::build => false,
+        }
+    }
+
+    pub fn uri(&self) -> String {
+        match self {
+            EndPoint::account => format!("account"),
+            EndPoint::account_achievements => format!("account/achievements"),
+            EndPoint::account_dailycrafting => format!("account/dailycrafting"),
+            EndPoint::account_dungeons => format!("account/dungeons"),
+            EndPoint::account_dyes => format!("account/dyes"),
+            EndPoint::achievements(op_id) => match op_id {
+                Some(id) => format!("achievements/{}", id.0.to_string()),
+                None => format!("achievements"),
+            },
+            EndPoint::achievements_daily => "achievements/daily".to_string(),
+            EndPoint::achievements_daily_tomorrow => "achievements/daily/tomorrow".to_string(),
+            EndPoint::account_materials => "account/materials".to_string(),
+            EndPoint::account_bank => "account/bank".to_string(),
+            EndPoint::items(id) => format!("items/{}", id.0.to_string()),
+            EndPoint::item_stats(op_stats_id) => match op_stats_id {
+                Some(stats_id) => format!("itemstats/{}", stats_id.0.to_string()),
+                None => "itemstats".to_string(),
+            },
+            EndPoint::item_stats_all(op_item_stat_id) => match op_item_stat_id {
+                Some(item_stat_id) => format!("itemstats/{}", item_stat_id.0.to_string()),
+                None => "itemstats".to_string(),
+            },
+            EndPoint::recipes(op_recipe_id) => match op_recipe_id {
+                Some(recipe_id) => format!("recipes/{}", recipe_id.0.to_string()),
+                None => "recipes".to_string(),
+            },
+            EndPoint::build => "build".to_string(),
+        }
+    }
+}
+
+pub struct Requester {
```

Comment:
This type is named Requester yet makes no requests.  What is the goal of this type?  It should be documented so it can be discussed.

Currently it seems that this type may not be needed.  The only value of its state is specifying a different base url or API version.  The base url may have some value, especially for testing.  Although to me it makes more sense that the dependency injection happen at a much higher layer then uri building.  The API version will never be changeable without huge changes to the library (assuming that a new version of the API ever happens).

If eventually it will be making requests then indicate so.  There is a lot of design work to talk about in this area.

Reply:
Could you elaborate on the dependency injection statement? What dependency are you talking about  the base url? I'd picture that the version if accounted for in the beginning of the libraries design would mean that there would be little to zero library changes. 

I do doubt they'll change the versions however. 
The documentation is certainly needed along with the intents, no doubt bout that.